preflight.inc.php

```diff
@@ -142,14 +142,21 @@
 		$errors++;
 	}
 
-	if(in_array('mod_rewrite', apache_get_modules())){
-		$tests['mod_rewrite']['state']="good";
-		$tests['mod_rewrite']['message']='mod_rewrite detected';
-		$tests['api_test']['state']="fail";
-		$tests['api_test']['message']="Apache does not appear to be rewriting URLs correctly. Check your AllowOverride directive and change to 'AllowOverride All'";
-	}else{
-		$tests['mod_rewrite']['state']="fail";
-		$tests['mod_rewrite']['message']='Apache is missing the <a href="http://httpd.apache.org/docs/current/mod/mod_rewrite.html">mod_rewrite</a> module and it is required for the API to function correctly.  Please install it.';
+    if (function_exists('apache_get_modules')) {
+        if (in_array('mod_rewrite', apache_get_modules())) {
+            $tests['mod_rewrite']['state']="good";
+            $tests['mod_rewrite']['message']='mod_rewrite detected';
+            $tests['api_test']['state']="fail";
+            $tests['api_test']['message']="Apache does not appear to be rewriting URLs correctly. Check your AllowOverride directive and change to 'AllowOverride All'";
+        } else{
+            $tests['mod_rewrite']['state']="fail";
+            $tests['mod_rewrite']['message']='Apache is missing the <a href="http://httpd.apache.org/docs/current/mod/mod_rewrite.html">mod_rewrite</a> module and it is required for the API to function correctly.  Please install it.';
+        }
+    } else {
```

Comment:
Instead of just assuming that mod_rewrite is present, do you have access to the httpd executable?  Where we can maybe execute a shell command like httpd -t -D DUMP_MODULES  or apache2 -t -D DUMP_MODULES  and then check that for the mod_rewrite?